.github/workflows/spack-matrix.yml

```diff
@@ -0,0 +1,136 @@
+name: spack-matrix
+
+on:
+  push:
+  #   branches:
+  #     - main
+  pull_request:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 2 * * MON" # weekly, mondays 2am
+
+jobs:
+  fenics-basix:
+    runs-on: ubuntu-latest
+    container: ubuntu:24.04
+    strategy:
+      fail-fast: false
+      matrix:
+        version: ["main", "0.10", "0.9", "0.8"]
+        variant: ["", "build_type=Debug", "build_type=Release"]
```

Comment:
Just go with Developer - it contains -g anyway and we will be turning on more and more strict checks.